dataclients/kubernetes/clusterclient.go

```diff
@@ -462,12 +493,18 @@ func (c *clusterClient) fetchClusterState() (*clusterState, error) {
 		return nil, err
 	}
 
+	secrets, err := c.loadSecrets()
```

Comment:
We should not load secrets if certificate registry is not used

Reply:
Now making sure secrets are not loaded if cert registry is not enabled.